corehq/apps/userreports/specs.py

```diff
@@ -39,29 +46,62 @@ def filter_from_spec(self, spec):
     @property
     @memoized
     def named_filters(self):
-        extra_filters = {}
-        if self.domain and toggles.UCR_EXPRESSION_REGISTRY.enabled(self.domain):
-            from corehq.apps.userreports.models import UCRExpression
-            extra_filters = UCRExpression.objects.get_wrapped_filters_for_domain(self.domain, self)
-        return extra_filters | self._named_filters
+        raise Exception("Use get_named_filter instead of named_filters directly.")
 
     @named_filters.setter
     def named_filters(self, named_filters):
         self._named_filters = named_filters
 
     @property
     @memoized
-    def named_expressions(self):
-        extra_expressions = {}
+    def _extra_filters(self):
+        extra_filters = {}
         if self.domain and toggles.UCR_EXPRESSION_REGISTRY.enabled(self.domain):
             from corehq.apps.userreports.models import UCRExpression
-            extra_expressions = UCRExpression.objects.get_wrapped_expressions_for_domain(self.domain, self)
-        return extra_expressions | self._named_expressions
+            extra_filters = UCRExpression.objects.get_wrapped_filters_for_domain(self.domain, self)
+        return extra_filters
+
+    def get_named_filter(self, name):
+        return self._get_named(name, self._named_filters | self._extra_filters, self._filter_stack)
```

Comment:
I notice this flips the order around the `|`, meaning `_extra_filters` will now override `_named_filters` where they conflict.  Was that change deliberate?

Reply:
6b19be13f442c7ed44312517c28786a39816a9f9